crates/hyperdrive-math/src/short/open.rs

```diff
@@ -34,19 +34,35 @@ impl State {
         spot_price: FixedPoint,
         mut open_vault_share_price: FixedPoint,
     ) -> Result<FixedPoint> {
+        if short_amount < self.config.minimum_transaction_amount.into() {
+            // TODO would be nice to return a `Result` here instead of a panic.
+            panic!("MinimumTransactionAmount: Input amount too low");
+        }
+
         // If the open share price hasn't been set, we use the current share
         // price, since this is what will be set as the checkpoint share price
         // in the next transaction.
         if open_vault_share_price == fixed!(0) {
             open_vault_share_price = self.vault_share_price();
         }
 
+        // TODO solidity uses mulUp here, should we do that here as well?
+        let share_reserves_delta_in_shares =
+            self.vault_share_price() * self.short_principal(short_amount)?;
```

Comment:
Looks like it was changed to round up a couple months ago. We should probably change this and the calc below to round up to remain at parity and overestimate the proceeds for the negative interest check.

https://github.com/delvtech/hyperdrive/pull/736/files#diff-114831fb0435b556f3b41af29e9f3b9e808cbd0dbd8d3ca504492a09ed4fd6fcR425

Also, I think the name of this variable is a typo, because the number is denominated in base.